randovania/server/multiplayer/web_api.py

```diff
@@ -98,13 +99,38 @@ def admin_session(user, session_id):
         )
 
     header = ["User", "World", "Connection State", "Inventory"]
-
-    return "<table border='1'><tr>{}</tr>{}</table>".format(
+    table = "<table border='1'><tr>{}</tr>{}</table>".format(
         "".join(f"<th>{h}</th>" for h in header),
         "".join("<tr>{}</tr>".format("".join(f"<td>{h}</td>" for h in r)) for r in rows),
     )
 
+    entries = [
+        f"<p>Session: {session.name}</p>",
+        f"<p>Created by {session.creator.name} at {session.creation_datetime}</p>",
+        "<p><a href='{link}'>Download rdvgame</a></p>".format(
+            link=flask.url_for("download_session_spoiler", session_id=session_id)
+        )
+        if session.has_layout_description()
+        else "<p>No rdvgame attached</p>",
+        table,
+    ]
+
+    return "\n".join(entries)
+
+
+def download_session_spoiler(user: User, session_id: int):
```

Comment:
mypy

Reply:
I actually didn't because I don't know the proper typing for flask functions. And can't really be bothered to find it.

Reply:
actually looked it up lol